### PR TITLE
Issue408: Add gh-pages support

### PR DESCRIPTION
**Pull request by mrosanes**
Add gh-pages support to be able to use
the dockers taurus-test for building documentation, 
and be able to get rid of mocks for building the docs 
later on.

Edit: This is a proposal to deal with #408 . The results can be seen in https://mrosanes.github.io/taurus/

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,7 +20,19 @@ matrix:
 before_install:
   - docker run -d --name=taurus-test -h taurus-test --volume=`pwd`:/taurus $DOCKER_IMG
   - sleep 10
+  - docker exec taurus-test /bin/bash -c "apt-get -qq update"
+  - docker exec taurus-test /bin/bash -c "apt-get install -y python-sphinx-rtd-theme"
```
Comment (cpascual) [inline .travis.yml]: I would do this (install sphinx-rtd-image) in the docker images instead.

File: .travis.yml [hunk 2]
```diff
@@ -20,7 +20,19 @@ matrix:
...
   
 script:
   - docker exec taurus-test /bin/bash -c "cd taurus ; python setup.py install"
-  - docker exec taurus-test /bin/bash -c "TAURUS_STARTER_WAIT=5 taurustestsuite -e 'taurus\.core\.util\.test\.test_timer'"
+  - travis_wait docker exec taurus-test /bin/bash -c "TAURUS_STARTER_WAIT=5 taurustestsuite -e 'taurus\.core\.util\.test\.test_timer'"
+  - docker exec taurus-test /bin/bash -c "cd taurus ; python setup.py build_sphinx"
+  - docker exec taurus-test /bin/bash -c "cd taurus/build/sphinx/html ; touch .nojekyll"
```
Comment (cpascual) [inline .travis.yml]: Maybe it would be better to avoid the `cd`: 
```
docker exec taurus-test /bin/bash -c "touch /taurus/build/sphinx/html/.nojekyll"
```

**Comment by cpascual** [comment]
Make sure that the deploy only happens with one image (I suggest debian-stretch).
This can be done [using `condition`](https://docs.travis-ci.com/user/deployment/#Conditional-Releases-with-on%3A):



Also, currently the

**Comment by cpascual** [comment]
Note that the resulting docs lack the graphs created with graphviz (it may be related to a missing dependency in the images). Compare, e.g.: https://mrosanes.github.io/taurus/devel/api/taurus/qt/qtgui/display/_TaurusLabel.html#tauruslabel with http://www.taurus-scada.org/en/latest/devel/api/taurus/qt/qtgui/display/_TaurusLabel.html#tauruslabel

File: .travis.yml (hunk 1 above)
Comment (mrosanes) [reply .travis.yml]: I agree, but I don't know if I have access to modify the taurus-test docker files. @cpascual, could you add the python-sphinx-rtd-theme to the dockers?

Comment (cpascual) [reply .travis.yml]: I just added it to debian-stretch and debian-buster images.
I am not sure we should add it to debian-jessie

**Comment by cpascual** [comment]
This approach was good as a proof of concept, but it has several issues:

- It uses a personal token (which would grant write travis access to *all* repos to which the user has access) instead of a deploy key
- It deploys to the same repo (again, a security problem)
- It does not support version-control of the docs nor publishing more than one version of the docs

These issues have been addressed in #572. Therefore I am closing this PR and we can continue the work on #572 and #408

**Pull request closed without merging**